run_windows.go

```diff
@@ -0,0 +1,527 @@
+		si          startupInfoEx
+	)
+
+	si.startupInfo.Cb = uint32(unsafe.Sizeof(si))
+
+	// get size of attr list
+	r1, _, err := initializeProcThreadAttributeList.Call(
+		0, // list ptr
+		1, // list item count
+		0, // dwFlags: reserved, MUST be 0
+		uintptr(unsafe.Pointer(&attrBufSize)),
+	)
+	if r1 == 0 {
+		// according to
+		// https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-initializeprocthreadattributelist
+		// which says: This initial call will return an error by design. This is expected behavior.
+		//
+		// so here we check the returned value of the attr buf size, if it's zero, we cannot update attribute list
+		if attrBufSize == 0 {
+			return nil, os.NewSyscallError("InitializeProcThreadAttributeList (size)", err)
+		}
+	}
+
+	attrListBuf := make([]byte, attrBufSize)
+	si.lpAttrList = syscall.Handle(unsafe.Pointer(&attrListBuf[0]))
+	// create attr list with console handler
+	r1, _, err = initializeProcThreadAttributeList.Call(
+		uintptr(si.lpAttrList),                // attr list buf
+		1,                                     // list item count
+		0,                                     // dwFlags: reserved, MUST be 0
+		uintptr(unsafe.Pointer(&attrBufSize)), // size of the list
+	)
+	if r1 == 0 {
+		// false
+		return nil, os.NewSyscallError("InitializeProcThreadAttributeList (create)", err)
+	}
+
+	// https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute
+	r1, _, err = updateProcThreadAttribute.Call(
+		uintptr(si.lpAttrList), // buf list
+		0,                      // dwFlags: reserved, MUST be 0
+		_PROC_THREAD_ATTRIBUTE_PSEUDOCONSOLE,
+		uintptr(consoleHandle),
+		unsafe.Sizeof(consoleHandle),
+		0,
+		0,
+	)
+	if r1 == 0 {
+		// false
+		_, _, _ = deleteProcThreadAttributeList.Call(uintptr(si.lpAttrList))
+		return nil, os.NewSyscallError("UpdateProcThreadAttribute", err)
+	}
+
+	return &si, nil
+}
+
+// copied from os/exec.(*Cmd).Start
+// start starts the specified command but does not wait for it to complete.
+//
+// If Start returns successfully, the c.Process field will be set.
+//
+// The Wait method will return the exit code and release associated resources
+// once the command exits.
+func start(c *cmd, consoleHandle syscall.Handle) error {
+	if c.lookPathErr != nil {
+		c.closeDescriptors(c.closeAfterStart)
+		c.closeDescriptors(c.closeAfterWait)
+		return c.lookPathErr
+	}
+	if runtime.GOOS == "windows" {
+		lp, err := lookExtensions(c.Path, c.Dir)
+		if err != nil {
+			c.closeDescriptors(c.closeAfterStart)
+			c.closeDescriptors(c.closeAfterWait)
+			return err
+		}
+		c.Path = lp
+	}
+	if c.Process != nil {
+		return errors.New("exec: already started")
+	}
+	if c.ctx != nil {
+		select {
+		case <-c.ctx.Done():
+			c.closeDescriptors(c.closeAfterStart)
+			c.closeDescriptors(c.closeAfterWait)
+			return c.ctx.Err()
+		default:
+		}
+	}
+
+	//c.childFiles = make([]*os.File, 0, 3+len(c.ExtraFiles))
```

Comment:
Please remove the unused commented code, or if it is kept for reference, please add a comment on why it is disabled and add a space after the double slash (no space being used for directive)

Reply:
same as https://github.com/creack/pty/pull/109#discussion_r607920765